server/repository/src/db_diesel/temperature_chart/temperature_chart_row.rs

```diff
@@ -0,0 +1,224 @@
+        //  UNION SELECT '2021-01-01T17:00:00' as from_datetime, '2021-01-01T18:00:00' as to_datetime
+        //  UNION SELECT '2021-01-01T18:00:00' as from_datetime, '2021-01-01T19:00:00' as to_datetime
+        //  ) AS time_series
+        //  JOIN temperature_log ON
+        //       (temperature_log.datetime >= time_series.from_datetime
+        //       AND temperature_log.datetime < time_series.to_datetime)
+
+        let mut union_prefix = "";
+
+        out.push_sql(" ( ");
+        for Interval {
+            from_datetime: from_date,
+            to_datetime: to_date,
+        } in self.intervals.iter()
+        {
+            out.push_sql(union_prefix);
+            // Only add UNION after first select
+            union_prefix = " UNION ";
+
+            out.push_sql(" SELECT ");
+            out.push_bind_param::<Timestamp, _>(&from_date)?;
+            out.push_sql(" as from_datetime, ");
+            out.push_bind_param::<Timestamp, _>(&to_date)?;
+            out.push_sql(" as to_datetime ");
+        }
+
+        out.push_sql(
+            r#" 
+                ) AS time_series
+               JOIN temperature_log ON 
+                    (temperature_log.datetime >= time_series.from_datetime
+                    AND temperature_log.datetime < time_series.to_datetime)"#,
+        );
+
+        Ok(())
+    }
+}
+
+impl QuerySource for TemperatureChart {
+    type FromClause = Self;
+    type DefaultSelection = AllColumns;
+    fn from_clause(&self) -> Self::FromClause {
+        // In expanded macro this Identifier("table_name"), which is translates to `table_name`
+        // in walk_ast. We change this to wal_ast of TemperatureLog
+        self.clone()
+    }
+    fn default_selection(&self) -> Self::DefaultSelection {
+        Self::all_columns()
+    }
+}
+
+// Boilerplate
+type SqlType = (Timestamp, Timestamp, Double, Text, Text);
+type AllColumns = (
+    FromDatetime,
+    ToDatetime,
+    AverageTemperature,
+    TemperatureLogId,
+    SensorId,
+);
+impl Table for TemperatureChart {
+    type PrimaryKey = FromDatetime;
+    type AllColumns = AllColumns;
+    fn primary_key(&self) -> Self::PrimaryKey {
+        FromDatetime
+    }
+    fn all_columns() -> Self::AllColumns {
+        (
+            FromDatetime,
+            ToDatetime,
+            AverageTemperature,
+            TemperatureLogId,
+            SensorId,
+        )
+    }
+}
+impl AppearsInFromClause<TemperatureChart> for TemperatureChart {
+    type Count = Once;
+}
+impl AppearsInFromClause<TemperatureChart> for () {
+    type Count = Never;
+}
+// pub type BoxedQuery<'a, DB, ST = SqlType> = BoxedSelectStatement<'a, ST, TemperatureChart, DB>;
+
+impl AsQuery for TemperatureChart {
+    type SqlType = SqlType;
+    type Query = SelectStatement<Self>;
+    fn as_query(self) -> Self::Query {
+        SelectStatement::simple(self)
+    }
+}
+
+// This is just boilerplate taken from expanding table! macro
+macro_rules! temperature_chart_column {
```

Comment:
I put this into macro, since it's quite verbose, and really hard to see for all of the columns